app/controllers/topics_controller.rb

```diff
@@ -0,0 +1,38 @@
+class TopicsController < ApplicationController
+  def index
+    @topics = Topic.all
+  end
+
+  def show
+    @topic = Topic.find(params[:id])
+  end
+
+  def new
+    @topic = Topic.new
+  end
+
+  def create
+    @topic = Topic.new
+    @topic.title = params[:topic][:title]
```

Comment:
This is fine because you're not doing mass assignment. However, as a matter of preference, I like the Strong Parameters approach. You'd extract a method like this:

```
def topic_params
  params.require(:topic).permit(:title)
end
```

And then use it in the create method like this:

```
@topic = Topic.new(topic_params)
# ...
if @topic.save
# ...
```